pkg/util/validation/label_value_length_over_limit_strategy_test.go

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package validation
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+func TestLabelValueLengthOverLimitStrategy_Unmarshal(t *testing.T) {
+	testCases := []struct {
+		input   string
+		want    LabelValueLengthOverLimitStrategy
+		wantErr bool
+	}{
+		{
+			input:   "invalid",
+			wantErr: true,
+		},
+		{
+			input: "",
+			want:  LabelValueLengthOverLimitStrategyError,
+		},
+		{
+			input: "error",
+			want:  LabelValueLengthOverLimitStrategyError,
+		},
+		{
+			input: "truncate",
+			want:  LabelValueLengthOverLimitStrategyTruncate,
+		},
+		{
+			input: "drop",
+			want:  LabelValueLengthOverLimitStrategyDrop,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run("input="+tc.input, func(t *testing.T) {
+			for _, format := range []struct {
+				name string
+				set  func(*LabelValueLengthOverLimitStrategy, string) error
+			}{
+				{name: "json", set: func(v *LabelValueLengthOverLimitStrategy, input string) error {
+					b, _ := json.Marshal(input)
```

Comment:
Errors shouldn't be ignored:
```suggestion
				{name: "json", set: func(t *testing.T, v *LabelValueLengthOverLimitStrategy, input string) error {
					t.Helper()
					b, err := json.Marshal(input)
					require.NoError(t, err)
```